pkg/cloudevents/transport/httpb/transport.go

```diff
@@ -0,0 +1,327 @@
+package httpb
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/bindings/http"
+	cecontext "github.com/cloudevents/sdk-go/pkg/cloudevents/context"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/plugin/ochttp/propagation/tracecontext"
+	"go.uber.org/zap"
+	"io"
+	"net"
+	nethttp "net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	"github.com/cloudevents/sdk-go/pkg/event"
+)
+
+// Transport adheres to transport.Transport.
+var _ transport.Transport = (*Transport)(nil)
+
+const (
+	// DefaultShutdownTimeout defines the default timeout given to the http.Server when calling Shutdown.
+	DefaultShutdownTimeout = time.Minute * 1
+)
+
+// Transport acts as both a http client and a http handler.
+type Transport struct {
+	sender   binding.Sender
+	receiver *http.Receiver // implements binding.Receiver
+
+	consumer transport.Receiver
+
+	// ShutdownTimeout defines the timeout given to the http.Server when calling Shutdown.
+	// If nil, DefaultShutdownTimeout is used.
+	ShutdownTimeout *time.Duration
+
+	// Port is the port to bind the receiver to. Defaults to 8080.
+	Port *int
+	// Path is the path to bind the receiver to. Defaults to "/".
+	Path string
+
+	// Receive Mutex
+	reMu sync.Mutex
+	// Handler is the handler the http Server will use. Use this to reuse the
+	// http server. If nil, the Transport will create a one.
+	Handler           *nethttp.ServeMux
+	listener          net.Listener
+	server            *nethttp.Server
+	handlerRegistered bool
+}
+
+// Option is the function signature required to be considered an http.Option.
+type Option func(*Transport) error
+
+func New(opts ...Option) (*Transport, error) {
+	t := &Transport{}
+	if err := t.applyOptions(opts...); err != nil {
+		return nil, err
+	}
+
+	if t.sender == nil {
+		client := nethttp.DefaultClient
+		target, _ := url.Parse("http://localhost:8080")
+		t.sender = http.NewSender(client, target)
+	}
+
+	return t, nil
+}
+
+func (t *Transport) applyOptions(opts ...Option) error {
+	for _, fn := range opts {
+		if err := fn(t); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Send implements Transport.Send
+func (t *Transport) Send(ctx context.Context, e event.Event) (context.Context, *event.Event, error) {
+	msg := binding.EventMessage(e)
+
+	if err := t.sender.Send(ctx, msg); err != nil {
+		return ctx, nil, err
+	}
+
+	return ctx, nil, nil
+}
+
+// SetReceiver implements Transport.SetReceiver
+func (t *Transport) SetReceiver(r transport.Receiver) {
+	t.consumer = r
+}
+
+// SetConverter implements Transport.SetConverter
+func (t *Transport) SetConverter(c transport.Converter) {
```

Comment:
The converter handling IMO should be implemented directly by `binding.BindingTransport`: https://github.com/cloudevents/sdk-go/issues/315

Reply:
This is not that kind of converter, it allows you to implement a try again in the client side for custom converters